gtk-rust-modules/src/hunter.rs

```diff
@@ -0,0 +1,86 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+mod module_common;
+use module_common::*;
+
+#[no_mangle]
+pub extern fn set_shared(ro_ptr: cptr, _ro_len: i32, rw_ptr: cptr, _rw_len: i32) {
+    let mut guard = CTX.lock().expect("Failed to aquire ctx lock");
+    let ctx = &mut (*guard);
+    unsafe {
+        ctx.replace(
+            Context {
+                grid: Box::from_raw(ro_ptr as _),
+                hunter: Box::from_raw(rw_ptr as _),
+                runners: Box::from_raw(rw_ptr.add(std::mem::size_of::<Hunter>()) as _),
+            }
+            );
```

Comment:
Indent :)

Reply:
Weird, my editor (not even cargo's formatter) moves it there and I have no idea why.... Overridden it but... confused as always :)